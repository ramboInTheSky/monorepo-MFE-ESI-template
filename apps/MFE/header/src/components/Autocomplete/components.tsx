import styled from "styled-components"

export const Li = styled.li`
    a {
        color: ${props => props.theme.colours.text.default};
        text-decoration: none;
        padding: 0.375rem 0;
        margin: 0.125rem 0;
        display: inline-block;
        font-size: 0.875rem;
        letter-spacing: 0.12px;
        line-height: 1.5;

        > span:first-child {
            text-transform: capitalize;
        }

        &:hover {
            text-decoration: underline;
        }
    }
`
export const Ul = styled.ul`
    margin: 0;
    padding: 0;
    list-style: none;
`
