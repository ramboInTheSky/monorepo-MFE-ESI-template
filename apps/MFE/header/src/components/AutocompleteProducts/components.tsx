import styled from "styled-components"
import Typography from "@mui/material/Typography"
import {breakpoints} from "@monorepo/themes"
import {ProductsMaxItemsData} from "../../models/features/searchBar"

type UlProps = {
    maxitems: ProductsMaxItemsData
}

export const Container = styled.div`
    span {
        font-size: 0.875rem;
        letter-spacing: 0.12px;
        line-height: 1.5;
        display: inline-block;

        > strong {
            font-family: ${props =>
                `${props.theme.colours.font.primary.medium.family}, ${props.theme.colours.font.default}`};
            text-transform: capitalize;
        }

        @media (min-width: ${breakpoints.values.md}px) {
            padding-top: 0.375rem;
            margin-top: 0.125rem;
        }
    }
`

export const Div = styled.div`
    display: flex;
    justify-content: flex-end;
    color: ${props => props.theme.colours.text.hyperlink};
    text-decoration: underline;
    font-size: 0.875rem;
    cursor: pointer;

    @media (min-width: ${breakpoints.values.md}px) {
        text-transform: ${props => props.theme.styles.AutoCompleteButton?.xs.textTransform};
    }

    @media (min-width: ${breakpoints.values.md}px) {
        display: ${props => props.theme.styles.AutoCompleteButton?.md.display};
        text-decoration: ${props => props.theme.styles.AutoCompleteButton?.md.decoration};
        color: ${props => props.theme.colours.header.autoCompleteButton.color};
        background-color: ${props => props.theme.colours.header.autoCompleteButton.backgroundcolor};
        padding: ${props => props.theme.styles.AutoCompleteButton?.md.padding};
        border-radius: ${props => props.theme.colours.header.autoCompleteButton.radius};
        text-transform: ${props => props.theme.styles.AutoCompleteButton?.md.textTransform};
    }
`
export const ButtonContainer = styled.div`
    text-align: ${props => props.theme.styles.AutoCompleteButton?.md.align};
`
export const Li = styled.li`
    && {
        flex-grow: 0;
        padding: 1rem;
        padding-left: 0;

        &:hover {
            text-decoration: underline;
        }

        a {
            color: ${props => props.theme.colours.text.default};
            text-decoration: none;
            margin: 0.125rem 0;
            display: inline-block;
            padding-top: 150%;
            margin: 0;
            height: 0px;
            position: relative;
            width: 100%;
            background: ${props => props.theme.colours.utilities.backgroundAccent};
            font-size: 0.875rem;
            letter-spacing: 0.12px;
            line-height: 1.5;

            img {
                position: absolute;
                top: 0;
                right: 0;
                left: 0;
                height: 100%;
                width: 100%;
            }

            &:hover {
                text-decoration: underline;
                padding-left: 0;
            }

            h5 {
                align-self: flex-start;
                margin: 0;
                font-family: ${props =>
                    `${props.theme.colours.font.primary.medium.family}, ${props.theme.colours.font.default}`};
            }
        }
    }
`

const hideNumberOfItems = maxItems => {
    const total = maxItems + 1
    return `n + ${total}`
}
export const Ul = styled.ul<UlProps>`
    margin: 0;
    padding: 0;
    list-style: none;

    display: flex;
    width: 100%;
    overflow-x: scroll;

    &::-webkit-scrollbar {
        display: none;
    }

    -ms-overflow-style: none;
    scrollbar-width: none;

    @media (min-width: ${breakpoints.values.md}px) {
        overflow-x: initial;
    }

    li {
        max-width: calc(1000px / (${props => props.maxitems.sm}));
        flex-basis: calc(1000px / (${props => props.maxitems.sm}));

        @media (max-width: ${breakpoints.values.sm - 1}px) {
            &:nth-child(${props => hideNumberOfItems(props.maxitems.xs)}) {
                display: none;
            }
        }
        @media (max-width: ${breakpoints.values.md - 1}px) {
            min-width: 125px;
        }

        @media (min-width: ${breakpoints.values.sm}px) and (max-width: ${breakpoints.values.md - 1}px) {
            max-width: calc(100% / ${props => props.maxitems.sm});
            flex-basis: calc(100% / ${props => props.maxitems.sm});

            &:nth-child(${props => hideNumberOfItems(props.maxitems.sm)}) {
                display: none;
            }
        }
        @media (min-width: ${breakpoints.values.md}px) and (max-width: ${breakpoints.values.lg - 1}px) {
            max-width: calc(100% / ${props => props.maxitems.md});
            flex-basis: calc(100% / ${props => props.maxitems.md});

            &:nth-child(${props => hideNumberOfItems(props.maxitems.md)}) {
                display: none;
            }
        }
        @media (min-width: ${breakpoints.values.lg}px) and (max-width: ${breakpoints.values.xl - 1}px) {
            max-width: calc(100% / ${props => props.maxitems.lg});
            flex-basis: calc(100% / ${props => props.maxitems.lg});

            &:nth-child(${props => hideNumberOfItems(props.maxitems.lg)}) {
                display: none;
            }
        }
        @media (min-width: ${breakpoints.values.xl}px) {
            max-width: calc(100% / ${props => props.maxitems.xl});
            flex-basis: calc(100% / ${props => props.maxitems.xl});

            &:nth-child(${props => hideNumberOfItems(props.maxitems.xl)}) {
                display: none;
            }
        }
    }
`

export const Title = styled(Typography)`
    && {
        white-space: nowrap;
        overflow: hidden;
        text-transform: capitalize;
        text-overflow: ellipsis;
        max-width: 100%;
        margin: 1rem 0 0;
    }
`
export const ProductDetailsContainer = styled.div`
    text-align: ${props => props.theme.styles.AutoCompleteButton?.xs.align};
`
