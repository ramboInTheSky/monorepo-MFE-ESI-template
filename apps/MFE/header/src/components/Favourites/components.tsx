import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export const Container = styled.div`
    display: flex;
    justify-content: center;
    align-content: center;

        a {
            display: flex;
            justify-content: center;
            margin: ${props => props.theme.styles.Favourites.xs.margin};
            height: ${props => props.theme.styles.Favourites.xs.height};
            padding: ${props => props.theme.styles.UpperHeader.xs.iconHitArea};

            @media (min-width: ${breakpoints.values.md}px) {
                height: ${props => props.theme.styles.Favourites.md?.height};

                margin: ${({theme: {styles}}) => {
                    return styles.Favourites.md?.margin || "0.625rem 0.625rem 0.5rem"
                }};
            }
        }
    }

    img {
        width: ${props => props.theme.styles.Favourites.img.xs.width};
        height: ${({theme: {styles}}) => styles.Favourites.img.xs.height};
        padding: ${props => props.theme.styles.Favourites.img.xs.padding};
    
        @media (min-width: ${breakpoints.values.md}px) {
            width: ${props => props.theme.styles.Favourites.img?.md?.width};
            height: ${props => props.theme.styles.Favourites.img?.md?.height};
        }
    }
`

export const Wrapper = styled.div`
    display: flex;
    align-items: center;
`
