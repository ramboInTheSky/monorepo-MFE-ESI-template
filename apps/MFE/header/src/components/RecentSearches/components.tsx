import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export const Li = styled.li`
    width: 100%;

    a {
        color: ${props => props.theme.colours.text.default};
        text-decoration: none;
        padding: 0.375rem 0.5rem 0.375rem 0;
        margin: 0.125rem 0;
        display: inline-block;
        font-size: 0.875rem;
        line-height: 1.5;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        width: 100%;
        @media (min-width: ${breakpoints.values.md}px) {
            width: 16.875rem;
        }
        :hover {
            text-decoration: underline;
        }
    }
`

export const Ul = styled.ul`
    @media (max-width: ${breakpoints.values.md}px) {
        width: 100%;
    }
`
