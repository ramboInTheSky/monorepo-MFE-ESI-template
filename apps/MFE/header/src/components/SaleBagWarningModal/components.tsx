import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export const ModalContainer = styled.div`
    position: absolute;
    top: 50%;
    left: 50%;
    margin-right: -50%;
    transform: translate(-50%, -50%);
    background: ${props => props.theme.colours.header.navUpperBackground.color};
    color: ${props => props.theme.colours.text.default};
    padding: 1.375rem 1.5rem;
    width: 90%;
    min-height: 16.625rem;
    @media (min-width: ${breakpoints.values.md}px) {
        min-height: 12.5rem;
        width: 80%;
    }
    @media (min-width: ${breakpoints.values.lg}px) {
        width: 55%;
    }
`

export const ButtonContainer = styled.div`
    height: 2.75rem;
    button:first-child {
        margin-bottom: 0.75rem;
    }

    @media (min-width: ${breakpoints.values.md}px) {
        float: right;
        width: 90%;
        display: flex;
        margin-top: 0.5rem;

        button:first-child {
            margin-right: 1rem;
            margin-bottom: 0;
        }
    }

    @media (min-width: ${breakpoints.values.lg}px) {
        height: 2.25rem;
        width: 80%;
    }
`
