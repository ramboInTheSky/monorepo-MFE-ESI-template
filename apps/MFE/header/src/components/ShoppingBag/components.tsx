import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export const ShoppingBagLink = styled.div`
    position: relative;
    display: flex;
    justify-content: center;
    align-content: center;
    height: ${props => props.theme.styles.ShoppingBag.xs.height};
    line-height: 1.125rem;
    @media (min-width: ${breakpoints.values.lg}px) {
        line-height: initial;
    }
    @media (min-width: ${breakpoints.values.md}px) {
        height: 2.8125rem;
    }
    && {
        i {
            font-style: normal;
            font-size: ${props => props.theme.colours.header.shoppingBag.fontSize};
            text-align: center;
            font-family: ${props =>
                `${props.theme.colours.font.primary.medium.family}, ${props.theme.colours.font.default}`};
            font-weight: ${props => props.theme.colours.header.shoppingBag.fontWeight};
            position: absolute;
            top: ${props => props.theme.styles.ShoppingBag.xs.top};
            left: ${props => props.theme.styles.ShoppingBag.xs.left};
            width: 100%;
            display: block;
            cursor: pointer;
            color: ${props => props.theme.colours.header.shoppingBag.color};

            @media (min-width: ${breakpoints.values.md}px) {
                top: ${props => props.theme.styles.ShoppingBag.md.top};
                font-size: 0.75rem;
            }
            @media (min-width: ${breakpoints.values.lg}px) {
                top: ${props => props.theme.styles.ShoppingBag.lg.top};
            }
        }

        img {
            width: ${props => props.theme.styles.ShoppingBag.xs.imageWidth};
            height: ${props => props.theme.styles.ShoppingBag.xs.imageHeight};

            @media (min-width: ${breakpoints.values.md}px) {
                width: ${props => props.theme.styles.ShoppingBag.md.imageWidth};
                height: ${props => props.theme.styles.ShoppingBag.md.imageHeight};
            }
        }

        a {
            padding: ${props => props.theme.styles.ShoppingBag.xs.padding};
        }

        button {
            padding: 0.625rem 0.625rem 0.5625rem 0.625rem;
            border: 0;
            background: 0;
        }
    }
`

export const ToolTipContent = styled.div`
    padding-top: 0.0625rem;
    border: ${props => props.theme.colours.popover.border};
    background: ${props => props.theme.colours.header.myAccount.background};
    border-radius: ${props => props.theme.colours.popover.radius};
    box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.4);
`
