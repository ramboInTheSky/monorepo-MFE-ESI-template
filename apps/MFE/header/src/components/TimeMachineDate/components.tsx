import styled from "styled-components"

export const Container = styled.div`
    background-color: red;
    position: absolute;
    z-index: 2;
    color: white;
    right: 0;
    top: 0;
    padding: 0.1rem;
`
