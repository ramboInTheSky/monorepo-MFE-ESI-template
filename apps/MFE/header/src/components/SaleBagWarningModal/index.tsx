import React, {useState} from "react"
import Typography from "@mui/material/Typography"
import CTA from "../CTA"

import Modal from "../Modal"
import connect from "./connect"
import {SaleBagWarning} from "../../models/text"
import {ModalContainer, ButtonContainer} from "./components"

type SaleBagWarningModalProps = {
    text: SaleBagWarning
    remainOnMainSiteAction: () => void
    vipSitePath: string
    openModal: boolean
}
export const SaleBagWarningModal = ({
    text: {title, textOne, textTwo, primaryButtonText, secondaryButtonText},
    remainOnMainSiteAction,
    openModal = false,
    vipSitePath,
}: SaleBagWarningModalProps) => {
    const [isOpen, setOpenModal] = useState(openModal)

    const remainOnMainSiteLogic = () => {
        remainOnMainSiteAction()
        setOpenModal(false)
    }

    const goToVipSite = () => {
        window.location.href = vipSitePath
    }
    return (
        <Modal open={isOpen} handleClose={remainOnMainSiteLogic} hideBackdrop={false}>
            <ModalContainer>
                <Typography variant="h3">{title}</Typography>
                <p>{textOne}</p>
                <p>{textTwo}</p>
                <ButtonContainer>
                    <CTA
                        testId="header_SaleBagWarningModal_goToVipButton"
                        enable
                        onClick={goToVipSite}
                        themeType="Secondary"
                        text={primaryButtonText}
                    />
                    <CTA
                        testId="header_SaleBagWarningModal_stayMainSiteButton"
                        enable
                        onClick={remainOnMainSiteLogic}
                        themeType="Primary"
                        text={secondaryButtonText}
                    />
                </ButtonContainer>
            </ModalContainer>
        </Modal>
    )
}

export default connect(SaleBagWarningModal)
