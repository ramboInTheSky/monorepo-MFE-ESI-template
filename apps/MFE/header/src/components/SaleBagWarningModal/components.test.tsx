import React from "react"
import {render} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {mockTheme} from "../../../__mocks__/mockStore"
import {ModalContainer, ButtonContainer} from "./components"

describe("SaleBagWarningModal component: ", () => {
    describe("ModalContainer: ", () => {
        it("should match the snapshot ", () => {
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <ModalContainer />
                </SCThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })
    describe("ButtonContainer: ", () => {
        it("should match the snapshot ", () => {
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <ButtonContainer />
                </SCThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })
})
