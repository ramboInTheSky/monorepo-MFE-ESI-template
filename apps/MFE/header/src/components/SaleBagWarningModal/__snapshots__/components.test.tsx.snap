// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaleBagWarningModal component:  ButtonContainer:  should match the snapshot  1`] = `
<DocumentFragment>
  .c0 {
  height: 2.75rem;
}

.c0 button:first-child {
  margin-bottom: 0.75rem;
}

@media (min-width:768px) {
  .c0 {
    float: right;
    width: 90%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    margin-top: 0.5rem;
  }

  .c0 button:first-child {
    margin-right: 1rem;
    margin-bottom: 0;
  }
}

@media (min-width:1024px) {
  .c0 {
    height: 2.25rem;
    width: 80%;
  }
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`SaleBagWarningModal component:  ModalContainer:  should match the snapshot  1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-right: -50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  background: #ffffff;
  color: #000000;
  padding: 1.375rem 1.5rem;
  width: 90%;
  min-height: 16.625rem;
}

@media (min-width:768px) {
  .c0 {
    min-height: 12.5rem;
    width: 80%;
  }
}

@media (min-width:1024px) {
  .c0 {
    width: 55%;
  }
}

<div
    class="c0"
  />
</DocumentFragment>
`;
